core/src/main/java/com/linecorp/armeria/common/HttpMessageAggregator.java

```diff
@@ -116,8 +116,11 @@ public void accept(Void unused, Throwable cause) {
             content = HttpData.of(merged);
         }
 
-        final AggregatedHttpMessage aggregated = onSuccess(content);
-        future.complete(aggregated);
+        try {
+            future.complete(onSuccess(content));
+        } catch (Exception e) {
```

Comment:
Maybe we should catch a `Throwable` here because otherwise the future will never be notified?